talisker/requests.py

```diff
@@ -302,3 +310,137 @@ def enable_requests_logging():  # pragma: nocover
     requests_log = logging.getLogger("requests.packages.urllib3")
     requests_log.setLevel(logging.DEBUG)
     requests_log.propagate = True
+
+
+class TaliskerAdapter(HTTPAdapter):
+
+    def __init__(self, backends=None, connect=1.0, read=10.0, max_retries=0,
+                 *args, **kwargs):
+        # set up backends
+        if backends:
+            if isinstance(backends, collections.Iterator):
+                self.backend_iter = backends
+            else:
+                backend_list = list(backends)
+                random.shuffle(backend_list)
+                self.backend_iter = itertools.cycle(backend_list)
+        else:
+            self.backend_iter = itertools.cycle([None])
+
+        self.connect_timeout = connect
+        self.read_timeout = read
+
+        if max_retries == 0:
+            self.__retry = None
+        elif isinstance(max_retries, int):
+            self.__retry = Retry.from_int(max_retries)
+        else:
+            self.__retry = max_retries
+
+        # disable lower level urllib3 retries completely
+        kwargs['max_retries'] = Retry(0, read=False)
+        super().__init__(*args, **kwargs)
+
+    def select_backend(self, request):
+        """Replaces the netloc of the url with a new netloc."""
+        next_endpoint = next(self.backend_iter)
```

Comment:
If we've passed in an iterator, this could fall off the end and raise an ugly exception.

Think we should catch this and raise a nicer error? Or `itertools.cycle` the given iterator instead to ensure that we don't fall off the end?